### PR TITLE
E2e test for conformance plugin

### LINKED ISSUE #3694: Verify conformance plugin e2e are working correctly
## Verify conformance plugin e2e are working correctly

This issue tracks work for verifying the state of E2E for the conformance plugin

### Verification

- [ ] Execute/run e2e tests when the conformance plugin source code is modified
- [ ] Is this workflow being triggered by the daily build?

### Issues found as a result of the audit?

We should prioritize these and determine if they are absolutely required for the workflow to function correctly and be reproducible.

- None yet

### PR DESCRIPTION
**Pull request by aman556**
## What this PR does / why we need it

This PR adds a E2E test for conformance Plugin.
- The flow of test is
	- Installing Unmanaged cluster.
	- Running conformance plugin.
	- Fetching the tar ball file which have the result of tarball.
	- Retrieve the result and print it.
	- Doing the cleanup of cluster, conformance and file.
- Small correction in path fields of Unmanaged and Diagnostic E2E test workflows.
## Which issue(s) this PR fixes

Fixes: #3694 

## Describe testing done for PR

- Run in Fork branch whose logs are https://github.com/aman556/community-edition/runs/6796153722?check_suite_focus=true
- Run `make conformance-e2e-test` in local.

### CODE REVIEW

File: cli/cmd/plugin/conformance/e2e-test/e2e_test.go
Comment (jpmcb) [inline cli/cmd/plugin/conformance/e2e-test/e2e_test.go:42]: This should use the `--tty-disable` flag to prevent the serialized emojis and stylized logging. The output will look much nicer

File: .github/workflows/e2e-conformance-plugin.yaml
Comment (jpmcb) [inline .github/workflows/e2e-conformance-plugin.yaml:50]: These conformance tests can take quite a long time.

Are we at risk of this timing out? Is there a timeout check we can put on this to keep it going for long periods of time (potentially hours)?

File: cli/cmd/plugin/conformance/e2e-test/e2e_test.go
Comment (jpmcb) [inline cli/cmd/plugin/conformance/e2e-test/e2e_test.go:72]: This all looks good to me but we should note that unmanaged cluster isn't targeted to be CNCF conformant and may have failures displayed in these results.

File: .github/workflows/e2e-conformance-plugin.yaml
Comment (aman556) [reply .github/workflows/e2e-conformance-plugin.yaml:50]: Hey @jpmcb Thanks for following up.
We are currently running the test for 180 minutes by running below command in `cli/cmd/plugin/conformance/Makefile`.
```
cd e2e-test && ROOT_DIR=$(ROOT_DIR) go test -tags=e2e -timeout 180m
```
I ran the test some more than 10 times and the time it took is around 90 minutes. So I think we are safe from timing out but yeah it’s not the idol step looking to set the -timeout flag for infinite(till the test Pass or fail) time.

File: cli/cmd/plugin/conformance/e2e-test/e2e_test.go
Comment (aman556) [reply cli/cmd/plugin/conformance/e2e-test/e2e_test.go:72]: Oh Okay so I am okay with not printing the result as that is not our main motive. What do you say?

File: .github/workflows/e2e-conformance-plugin.yaml
Comment (jpmcb) [reply .github/workflows/e2e-conformance-plugin.yaml:50]: Sounds great!!!

File: cli/cmd/plugin/conformance/e2e-test/e2e_test.go
Comment (jpmcb) [reply cli/cmd/plugin/conformance/e2e-test/e2e_test.go:72]: 🤷🏼 

It could be useful to validate that _some_ results are being printed correctly (since that's one of the big value adds of the conformance plugin). But it's not super relevant to ensuring it works. So it's up to you!

Comment (aman556) [reply cli/cmd/plugin/conformance/e2e-test/e2e_test.go:72]: Yeah, agree with the value add by the result. Let’s have it then.

File: cli/cmd/plugin/conformance/e2e-test/e2e_test.go
Comment (aman556) [reply cli/cmd/plugin/conformance/e2e-test/e2e_test.go:42]: Hey @jpmcb I add this flag but due to the way our cliRunner() function is printing the log we are not achieving the good way of Printing logs. I am investigating into this  will change once whenever found the root cause. For now I am adding a future task as comment in the code.
Thanks!

**Comment by aman556** [comment]
Hey @jpmcb can we please merge this if nothing major left. I will figure out that printing the log thing once other priority tasks get pop from stack.
Thanks!

**Comment by aman556** [comment]
Hey @stmcginnis can I please have your eyes on this one too. I think we are good to merge this one.
Thanks!

**Review by stmcginnis — APPROVED** [review]
Looks great!

**Comment by karuppiah7890** [comment]
@stmcginnis I see that this PR is approved. Do we merge this already? As I see the "Squash and merge" button. Are we waiting for more approvals?

**Pull request closed without merging**
